### PR TITLE
#235 new outbound flow 2

### PR DESCRIPTION
**Pull request by joshxg**
Really, mainly just moving a bunch of stuff around to reuse for the customer list.

- I took the components with their styling from the `useDialog` hook and put them in files. Wasn't sure if I should go all the way and just put all the sub components in files 🤷  I did this because I wanted to make a modal which doesn't have the bottom buttons but wanted to reuse the styles from the `Dialog` and `DialogTitle` even though there's not much there.
- I renamed things from the mui `Dialog` to `Modal` and I feel silly about it. I also did this for `TextField` and `TextInput`. I'm not sure why I did this.
- I originally was going to use the `ModalAutocomplete` component so I moved that into an `Autocomplete` from which I would build `ListAutocompelte`. But ended up being easier to just use the mui component
- Moved some of the `ModalAutocomplete` defaults into a shared file to reuse and also made a `TextInput` which has the small amount of styling to build on

### CODE REVIEW

File: packages/common/src/ui/components/inputs/Autocomplete/Autocomplete.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  height: 500,
+  padding: 10,
+  width: 400,
+});
+
+const options = Array.from({ length: 100 }).map((_, i) => ({
+  name: `${i}`,
+}));
+
+// TODO: Currently the styles are broken for this only within storybook
```
Comment (joshxg) [inline packages/common/src/ui/components/inputs/Autocomplete/Autocomplete.stories.tsx:24]: this is super annoying but theres some weird styling issues which I can't seem to fix 😢  Only showing up in storybook

File: packages/common/src/ui/components/inputs/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React, { PropsWithChildren } from 'react';
```
Comment (joshxg) [inline packages/common/src/ui/components/inputs/Autocomplete/Autocomplete.tsx:1]: This is the `ModalAutocomplete` which I've just taken a couple things and put them in types/utils to reuse them in `AutocompleteList`

File: packages/common/src/ui/components/inputs/Autocomplete/AutocompleteList.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
+import React from 'react';
```
Comment (joshxg) [inline packages/common/src/ui/components/inputs/Autocomplete/AutocompleteList.tsx:1]: styling for this one ended up being pretty annoying. Potentially would have been better off just using the `useAutocomplete` hook from mui. Still needs polish but I got a bit bored of styling so will revisit ..

File: packages/common/src/ui/components/modals/BasicModal/BasicModal.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          minHeight: `${height || '400'}px`,
+          minWidth: `${width || '500'}px`,
+        },
+        ...PaperProps,
+      }}
+      {...dialogProps}
+    >
+      {dialogProps.children}
+    </Dialog>
+  );
```
Comment (joshxg) [inline packages/common/src/ui/components/modals/BasicModal/BasicModal.tsx:29]: Just the `Dialog` you had in `useDialog` from which I wanted to use, but I wanted to have the control one level higher than the hook so ended up taking this out to use in a different places - again not really much styling but nice to not have to redo it

File: packages/common/src/ui/components/modals/ListSearch/ListSearch.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { ListSearch } from './ListSearch';
+
+export default {
+  title: 'Modals/ListSearch',
+  component: ListSearch,
+};
+
+const options = Array.from({ length: 100 }).map((_, i) => ({
+  name: `${i}`,
+}));
```
Comment (joshxg) [inline packages/common/src/ui/components/modals/ListSearch/ListSearch.stories.tsx:13]: Was thinking to maybe look at exporting some functions from `mock-server` to generate random data. Will also hook up a new story which would use some async data, so will revisit this there

File: packages/common/src/ui/components/modals/ListSearch/ListSearch.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+export const ListSearch = <T extends unknown>({
+  title,
+  options,
+  optionKey,
+  open,
+  onClose,
+}: ListSearchProps<T>): JSX.Element => {
+  const { height } = useWindowDimensions();
```
Comment (joshxg) [inline packages/common/src/ui/components/modals/ListSearch/ListSearch.tsx:23]: I'm not sure if this is a very good way to do this. I think maybe trying to flex generally always ends up being a better idea

File: packages/common/src/ui/components/modals/ListSearch/ListSearch.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  return (
+    <BasicModal open={open} onClose={onClose} height={modalHeight}>
+      <ModalTitle title={title} />
+      <AutocompleteList
+        options={options}
+        optionKey={optionKey}
+        height={listViewHeight}
+      />
+    </BasicModal>
```
Comment (joshxg) [inline packages/common/src/ui/components/modals/ListSearch/ListSearch.tsx:36]: Didn't add the button yet. Wasn't sure if we need to create customers on a remote anyway so maybe not a phase 1 thing

**Review by mark-prins — APPROVED** [review]
👍  LGTM!

that's awesome, thanks for tidying the autocompleter up, much better!

**Pull request merged**